examples/chain-of-density/chain_of_density.py

```diff
@@ -0,0 +1,223 @@
+    """
+
+    summary: str = Field(
+        ...,
+        description="This is a summary of the article provided which is overly verbose and has fillers to reach ~80 words",
+    )
+
+
+# Note that we utilise Spacy for entity recognition so that it is consistent with the original paper implementation which uses it as an original prompt
+class RewrittenSummary(instructor.OpenAISchema):
+    """
+    This is a new,denser summary of identical length which covers every entity and detail from the previous summary plus the missing entities provided.
+
+    Guidelines
+    - Make every word count : Rewrite the previous summary to improve flow and make space for additional entities
+    - Never remove an existing entity from the original text
+    - Make space with fusion, compression, and removal of uninformative phrases like "the article discusses"
+    - Missing entities can appear anywhere in the new summary
+    """
+
+    summary: str = Field(..., description="Concise yet self-contained summary")
+
+    @model_validator(mode="after")
+    def validate_sources(self, info: FieldValidationInfo) -> "RewrittenSummary":
+        original_summary = info.context.get("original_summary")
+
+        # We first extract out the original summary and compute the entity count
+        evaluation: MaybeOmittedEntities = openai.ChatCompletion.create(
+            model="gpt-4-0613",
+            messages=[
+                {
+                    "role": "system",
+                    "content": "You are about to be handed two summaries. Determine if the second summary has removed any entities that are present within the first summary",
+                },
+                {
+                    "role": "user",
+                    "content": f"Here is the first summary : {original_summary}",
+                },
+                {
+                    "role": "user",
+                    "content": f"Here is the second summary : {self.summary}",
+                },
+            ],
+            response_model=MaybeOmittedEntities,
+        )
+        if evaluation.omitted_entities:
+            omitted_entities = [
+                entity.entity_name for entity in evaluation.omitted_entities
+            ]
+            raise ValueError(
+                f"The following entities were omitted from the summary - {','.join(omitted_entities)}. Please generate a new summary which does not omit them."
+            )
+
+        return self
+
+
+def rewrite_summary(
+    article: str,
+    existing_summary: str,
+    entity_ctx: str,
+    error_msgs: List[str] = [],
+    remaining_retries=3,
+):
+    # # We then perform a new summary and validate that the entity density has increased ( We have not lost any entities )
+    completion = openai.ChatCompletion.create(
+        model="gpt-3.5-turbo-16k",
+        functions=[RewrittenSummary.openai_schema],
```

Comment:
Any reason response_model does not work here?

Reply:
Wanted to use the context validation here to validate the schema and the docs showed this method so I used it.

Reference: https://jxnl.github.io/instructor/examples/exact_citations/?h=fieldvalidation#validation-method-validate_sources

Reply:
ah i should clena up the docs to use `response_model` as much as possible

Reply:
Sure thing man.